### DISCUSSION TITLE
OpenSSL 3 Mac OSSL_LIB_CTX_load_config() loading openssl.cnf fails

### DISCUSSION THREAD

**Discussion by aniljadaun** [openssl/openssl#21611] [Q&A] [answered]
This code fails

    const char* default_search_path = "/Users/anils/openssl3legacyinbuilt/fipsdll";
        OSSL_PROVIDER_set_default_search_path(NULL, default_search_path);
        if (!OSSL_LIB_CTX_load_config(NULL, "Users/anils/openssl3legacyinbuilt/openssldir/openssl.cnf")) {
            FILE* fptr = NULL;
              if ((fptr = fopen("/Users/anils/log.txt", "a+")) != NULL) {
                   fprintf(fptr, "%s\n", "Error logging ::::");
                   ERR_print_errors_fp(fptr);
                   fclose(fptr);
              }
            
        }

------------------------------------------------------------------
OpenSSL libs configure command:

./Configure darwin64-arm64-cc enable-capieng no-module --prefix=/Users/anils/openssl3legacyinbuilt/arm64/dll/release --openssldir=/Users/anils/openssl3legacyinbuilt/openssldir

Error dumped from above command
--------------------------------------------------------
```
0046090002000000:error:80000002:system library:BIO_new_file:No such file or directory:crypto/bio/bss_file.c:67:calling fopen(Users/anils/openssl3legacyinbuilt/openssldir/openssl.cnf, rb)
0046090002000000:error:10000080:BIO routines:BIO_new_file:no such file:crypto/bio/bss_file.c:75:
0046090002000000:error:07000072:configuration file routines:def_load:no such file:crypto/conf/conf_def.c:179:
```
openssl.cnf
--------------------------------------------------------
```
config_diagnostics = 1
openssl_conf = openssl_init

.include "/Users/anils/openssl3legacyinbuilt/openssldir/fipsmodule.cnf"

[openssl_init]
providers = provider_sect


[provider_sect]
fips = fips_sect

[fips_sect]
identity=fips_test
```

please suggest what is the issue here. I have used no-module.  ?

Though i am able to load fips, legacy, default providers successfully. I am not sure what this error is in loading openssl.cnf file

**Comment by t8m** [comment, marked as answer]
It is clearly shown in the error message - the file `Users/anils/openssl3legacyinbuilt/openssldir/openssl.cnf` cannot be opened. Most probably because it is a relative path (missing `/` at the beginning).
